whitebox_inversion/utils.py

```diff
@@ -1,36 +1,42 @@
 import torch
 
 def to_relaxed_one_hot(tokens: torch.Tensor, vocab_size: int, hot_val: float = 1.0) -> torch.Tensor:
-    batch_size, seq_len = tokens.size()
-    one_hot = torch.zeros(batch_size, seq_len, vocab_size, device=tokens.device)
-    one_hot.scatter_(2, tokens.unsqueeze(-1).to(torch.int64), hot_val)
-    remaining_prob = hot_val / (vocab_size - 1)
-    one_hot += remaining_prob * (1 - one_hot)
-    return one_hot.to(tokens.device)
+    device = tokens.device
+    batch_size, seq_len = tokens.size()    
+    # Calculate the 'cold' value
+    cold_val = hot_val / (vocab_size - 1)    
+    # Create the tensor with the 'cold' value
+    one_hot = torch.full((batch_size, seq_len, vocab_size),
+                         fill_value=cold_val,
+                         device=device, dtype=torch.float32)    
+    # Use scatter to set the 'hot' values, factoring in normalisation condition
+    one_hot.scatter_(2, tokens.unsqueeze(-1), cold_val + (1.0-cold_val)*hot_val)    
+    return one_hot
 
 def simplex_projection(tensor: torch.Tensor) -> torch.Tensor:
-    d_batch, d_tokens, d_vocab = tensor.shape
+    batch, sequence, dims = tensor.shape
```

Comment:
I feel like i prefer the d_batch etc cause it dimensionality of the stuff etc - also dims is a bit charged cause "dim" is a flag - also dim in this case is the d_vocab

Reply:
oh yeah sorry this was just a copy-paste mistake